app/javascript/mastodon/reducers/compose.js

```diff
@@ -128,9 +128,7 @@ function appendMedia(state, media) {
     map.set('resetFileKey', Math.floor((Math.random() * 0x10000)));
     map.set('idempotencyKey', uuid());
 
-    const isSpoilerActive = enableAlwaysShowSpoiler ? state.get('spoiler_text').length > 0 : state.get('spoiler');
-
-    if (prevSize === 0 && (state.get('default_sensitive') || isSpoilerActive)) {
+    if (prevSize === 0 && (state.get('default_sensitive') || isSpoilerActive(state))) {
```

Comment:
I'd rather simplify this logic:
```suggestion
    if (prevSize === 0 && state.get('default_sensitive')) {
        map.set('sensitive', true);
    }
    if (isSpoilerActive(state)) {
        map.set('sensitive', true);
    }
```
The idea is: write the obvious logic, then enforce the isSpoilerActive constraint.

Reply:
That would change the behavior here.  We want to make sure we're changing states here.